### ISSUE TITLE
Preliminary Accessibility and QA Testing of iPhone Findings

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Child-Labor#18]
# New ILAB iPhone Application Issues

QA team started doing preliminary testing of the new ILAB iPhone application.

We have listed the issues by screen and if it is an accessibility issue, recommendation for usability improvement, question, etc. We still need to test the application with VoiceOver (iPhone screen reader).
## Home Screen
- [x] The poor color contrast of the greater than sign "**>**" which is used to get a person to the section they are trying to get to. _The disclosure indicator is system standard and not editable; you can see more information in the Human Interface Guidelines at_ https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/MobileHIG/ContentViews.html#//apple_ref/doc/uid/TP40006556-CH13-SW3 **Still think it's an issue even if Apple says it's fine. 09/09/2015**
- [x] The more information icon at bottom right of the screen is hard to see because of poor color contrast. _This has been adjusted in subsequent betas; the current appearance is:_ **Still think it's an issue even though it's better. 09/09/2015**
  ![Image of home screen](http://i.imgur.com/vCpBmYS.jpg =250x)
- [x] **QUESTION** - How come the line under the "Countries" and "Goods" navigation do not go all the way to the left as the other two lines do? _Item separator lines are system standard and not editable; in general, the outlines at the top and bottom of a table item group go all the way across, and lines in between any two items leave a margin stylistically._ **09/09/2015**
## More Information Screen
- [ ] **RECOMMENDATION** - Might be useful to have a glossary of terms page along with a short description of what each of the icons represents. _I'll bring it up internally, but our intention was never to make the icons totally unique or descriptive on their own, but just as quick way to view the type of good and then tap through for more information._
- [x] **RECOMMENDATION** - The "Done" link/button should be on the left side where most of the other back buttons are in other applications. _iOS convention is that "Back" buttons are placed on the left but "Done" buttons are placed on the right, the reason being that "Back" takes you to a previous navigation screen which slides in from the left, but "Done" closes a model which slides in from the bottom. There is more discussion of this available at_ http://globalmoxie.com/blog/ios-button-placement.shtml  **09/09/2015**
## Methodology Screen
- [x] The "**TDA report introduction**" links to a scanned PDF - http://www.dol.gov/ilab/reports/child-labor/findings/2013tda/2013tda.pdf Will have to double check it with JAWS or VoiceOver. Doing a quick review of this document, it seems to be a scanned PDF. Would this point to the newer 2014 version of the document? _Because this simply links to an external URL in Safari, we don't have control over this from the app, so it's_ **OUT OF SCOPE** _for the app._ **PDF links have been removed from the application. 09/09/2015**
- [x] We have the same issue as above with the "**TVPRA report Introduction**" PDF - http://www.dol.gov/ilab/reports/pdf/TVPRA_Report2014.pdf _Because this simply links to an external URL in Safari, we don't have control over this from the app, so it's_ **OUT OF SCOPE** _for the app._ **PDF links have been removed from the application. 09/09/2015**
- [x] We need to check the  PDF from the "**EO report introduction**" link to see if the PDF is accessible  - http://www.dol.gov/ilab/reports/pdf/EO_Report_2014.pdf _Because this simply links to an external URL in Safari, we don't have control over this from the app, so it's_ **OUT OF SCOPE** _for the app._ **PDF links have been removed from the application. 09/09/2015**
## Countries Screen
- [x] The poor color contrast of the greater than sign "**>**" which is used to get a person to the section they are trying to get too. _The disclosure indicator is system standard and not editable; more information is available in the Human Interface Guidelines at_ https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/MobileHIG/ContentViews.html#//apple_ref/doc/uid/TP40006556-CH13-SW3 **09/09/2015**
- [x] **RECOMMENDATION** - The greater than sign "**>**" could be moved to the left a bit so it's easier to tap on the letters to move the person down the list of countries. _The position and spacing of the disclosure indicator is system standard and not editable; more information is available in the Human Interface Guidelines at_ https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/MobileHIG/ContentViews.html#//apple_ref/doc/uid/TP40006556-CH13-SW3 **09/09/2015**
- [x] **RECOMMENDATION** - Should move the alphabetic list left a bit, so it's not right up against the right side of the screen. _The position of the section index relative to the disclosure indicators is system standard and not editable; more information is available in the Human Interface Guidelines at_ https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/MobileHIG/ContentViews.html#//apple_ref/doc/uid/TP40006556-CH13-SW3 **09/09/2015**
- [ ]  **RECOMMENDATION** - When a person scrolls down list of goods in the (A -Z, by level, and by sector) sections, they don't start at the top of the list or where they left off before. They end up down the page where the last scrolling was done in a prior section. _I'll investigate, but this seems to be standard behavior for a table view._
## Countries Screen - A - Z
- [x] The text of country "**Falkland Islands (Islas Malvinas)**" gets truncated when in portrait mode, along with about six others. _This is standard iOS behavior for table view items with long names in a table view. The user should be able to view the whole name by tapping through to the country's profile screen._ **09/14/2015**
## Countries Screen - By Level
- [x] There a few headings at the top of the page with no countries under them, along with most of them under the "**No Data**" heading. My guess is you are still working on these. _This was a bug in a prior beta; it's been corrected, and the countries are now distributed throughout the levels._ **09/14/2015**
## Individual Countries Screen
- [x] The poor color contrast of the greater than sign "**>**" which is used to get a person to the section they are trying to get too. _The disclosure indicator is system standard and not editable; more information is available in the Human Interface Guidelines at_ https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/MobileHIG/ContentViews.html#//apple_ref/doc/uid/TP40006556-CH13-SW3 **Still think it's an issue even if Apple says it's fine. 09/09/2015**
- [ ] The goods icons need their color contrast improved. It is hard to see them. Viewing them on their phone outside in bright sunlight will make this even more difficult. _This matches the appearance of goods in their other locations in the app for visual consistency, which has been governed by usability considerations. See the first bullet under Exploitation Type Screen for more information._
- [x] Unless a goods icon gets split on the right side, it is difficult to tell if there are more of goods in the carousel list. We should add previous and next arrows to make this easier to figure out unless the list only fills the screen. _This is standard behavior for a horizontally scrolling iOS collection view, similar to the behavior of any vertically scrolling page. iOS doesn't provide any option for indicators besides the collection view's own scroll bar—arrows would likely look too similar to the table view's disclosure icon, which indicates tappability instead of scrollability in the parlance of iOS._ **09/09/2015**
- [x] Text under the goods icons sometimes is not fully written out unless you go to landscape mode and even then some are not fully visible. _Similar to the standardized way that iOS table view items truncate long names, I constructed this collection view to do the same. The intention, as with Apple's own assumption on table views, is that a user can table to go to the good's profile screen to see the whole name._ **Still think it's an issue even if Apple says it's fine. 09/09/2015**
- [ ] Not sure best way to tell people that the goods icons can be tapped so see all countries that have issues with this particular good. Might want to add information about this in the "**More Info**" section. _A good suggestion, and I will run it by ILAB, who is generating the content for the "More Info" section._
- [ ] The little icons under the goods icons are hard to tell what they are. _The design is defined by the Exploitation Type Screen, see first bullet under that section for response._
- [x] **RECOMMENDATION** - Pages have the additional word "**Advancement**" on most or all individual country pages. _This has been corrected in subsequent betas._ **09/09/2015**
- [x] **RECOMMENDATION** - The "**More**" link at the end of the paragraph should be left justified, so it's easier to see and read for those with reading issues such as dyslexia. _This isn't a convention in the HIG, but a similar button appears in Apple's own App Store, and I was mimicking Apple's implementation—the placement on the right seems to be a visual indication that more text will be appended to the end._ **Still think it's an issue even if Apple says it's fine. 09/09/2015**
## Individual Countries - Suggested Actions Screen
- [x] The poor color contrast of the headings such as "**Laws**", "**Enforcement**", etc. Should probably make this text bold/strong so when you have no "**No Actions**" it's easier to tell the heading from the content. _These are system standard table view headings and footers, and we don't have control over their styling. "Laws," "Enforcement," etc., are standard table group headers styled by the operating system. "No Actions" is a standard table group footer styled by the operating system._ **Still think it's an issue even if Apple says it's fine. 09/09/2015**
- [x] The first item under "**Enforcement**"  for Brazil does not list the entire text even in landscape mode. Why not list everything that is available on the screen? _This has been corrected in subsequent betas._ **09/09/2015**
- [x]  Why no make the items under a heading an unordered/bulleted list. Doing this will help those using VoiceOver to get an idea of how many items are in the list. _iOS app layout does not use HTML; iOS's standard control for listing items is the table view._ **09/09/2015**
## Individual Countries - Statistics Screen
- [ ] The words under the heading "**Working**" have poor color contrast and are difficult to read even without bright sunlight. _UPDATE 9/13/15: Addressed by_ https://github.com/USDepartmentofLabor/Child-Labor/issues/42 **Color contrast doesn't look like it changed much if any. 09/14/2015**
## Individual Countries - International Conventions Screen
- [x] Poor color contrast of  the headings. _These are system standard table view headings and footers, and we don't have control over their styling._ **Still think it's an issue even if Apple says it's fine. 09/09/2015**
- [ ] The word "**YES**" that is in green has poor color contrast and should be made dark to say a forest green. _UPDATE 9/13/15: Addressed by_ https://github.com/USDepartmentofLabor/Child-Labor/issues/41 **Color contrast doesn't look like it changed much if any. 09/14/2015**
## Individual Countries - Legal Standards Screen
- [ ] The word "**YES**" that is in green has poor color contrast and should be made dark to say a forest green. _UPDATE 9/13/15: Addressed by_ https://github.com/USDepartmentofLabor/Child-Labor/issues/41 **Color contrast doesn't look like it changed much if any. 09/14/2015**
- [x] **QUESTION** - What are the additional numbers in parentheses on this screen? _UPDATE 9/13/15: These appear on legal standards referring to "minimum age"; this is the minimum age set by the country for that legal standard._ **09/14/2015**
- [ ] The text describing the age is not fully written out in portrait mode and could be confusing for those using the application that are not experts In information. _This has been corrected in future betas._ **In the latest version I have this issue has not been corrected. 09/14/2015**
## Individual Countries - Report PDF Screen
- [x] We need to make sure all the PDF's are readable to VoiceOver users. Plus, the text in them is way to small to read on the persons iPad let alone the iPhone. _These are being removed in the launch version of the app._ **09/09/2015**
## Goods Screen
- [x] The poor color contrast of the greater than sign "**>**" which is used to get a person to the section they are trying to get too. _The disclosure indicator is system standard and not customizable._ **Still think it's an issue even if Apple says it's fine. 09/09/2015**
## Goods - A - Z Screen
- [x] The poor color contrast of the greater than sign "**>**" which is used to get a person to the section they are trying to get too. _The disclosure indicator is system standard and not customizable._ **Still think it's an issue even if Apple says it's fine. 09/09/2015**
- [ ] **RECOMMENDATION** - Should add the alphabetic list to the right as on other pages so people can skip down the list to the section they want. _This is feasible, and would seem appropriate for this alphabetically-organized list. I'll bring this up with ILAB to see if they approve of this design recommendation._
## Goods - By Sector Screen
- [x] **RECOMMENDATION** - Should add the alphabetic list to the right as on other pages so people can skip down the list to the section they want. _This is a totally logical recommendation, and I mocked this up, but we decided that in practice, it doesn't really work for sections that are anything other than alphabetic. Instead of 26 one-letter sections that run neatly and narrowly down most of the side of the screen, you'd have only five or so very long section names.......iOS centers them vertically on the right side of the screen, pushing the rest of the content of the page left and creating massive whitespace along the entire right side of the screen above and below these 5 tiny options. We decided not to add a section index to any list that wasn't groups alphabetically._ **09/09/2015**
## Individual Goods Screen
- [ ] When in Portrait mode and a few cased in landscape mode the top heading is not fully visible. The heading should go to a second or third line if need be. Some people using the application might not be experts in all the goods or other information. Example - "**Bidis (hand-or**" shows up in portrait mode, while in landscape mode it reads "**Bidis (hand-rolled cigar...**" when the full heading is "**Bidis (hand-rolled cigarettes)**". _This would indeed be the desired behavior, but may not be fixable by submission on Monday. I'll try hard to find a fix before then._
- [ ] When selecting labor types such as "**Child Labor**" therefore I can only see all the other types that don't apply and not the grayed out ones. I would expect them not to be part of the list anymore. For an example see "**Bricks**" and tap on "**Forced Child Labor**" where 10 or so items are now grayed out. A good example is how on the "**Exploitation Types**" page when you tap on one labor type all those that do not apply disappear. _This was a design decision by ILAB, but I will ask them to revisit._
- [x] The little "**hand**" and "**people**" icons are hard to see because of their small size especially when there is more than one of them per country. _See third bullet under Exploitation Type Screen for design rationale._ **09/14/2015**
- [ ] **RECOMMENDATION** - Possibly add an alphabetic list to the longer lists that are more than a page. Might not need whole alphabetic list but only the letters needed. _This was a design decision by ILAB. I will ask them to revisit._
## Exploitation Type Screen
- [x] The goods icons need their color contrast improved. It is hard to see them. Viewing them on their phone outside in bright sunlight will make this even more difficult.
  _I’m concerned about increasing this too much, because the darker the goods are, the more the exploitation symbols on top of them get lost in them visually. See what happens when I increase the darkness of them by 25%:_ **Even with upping the color 25% it still doesn't pass the color contrast ratio of 4.5:1. This will make it even harder to see out in bright sun.** **Still think it's an issue even if Apple says it's fine. 09/09/2015**

![Image of exploitation screen](http://i.imgur.com/zQWlQwq.png)

_It becomes much harder to see that the symbols on top of them are even separate entities._

_I will tinker with making them a shade darker, perhaps 5%, and see if that helps their own visibility without obscuring the exploitation symbols. But in general, for the intended use cases of this screen, it’s not a priority that the user necessarily be able to visually identify every good. The most important use of this screen is to see which countries have the most goods and what types of exploitation dominate. Thus, I’m reluctant to increase the recognizability of the individual goods at the expense of the exploitation types attached to each._
- [x] Unless a goods icon gets split on the right side, it is difficult to tell if there are more of them on the list. We should add previous and next arrows to make this easier to figure out unless the list only fills the screen. _This is standard behavior for a horizontally scrolling iOS collection view, similar to the behavior of any vertically scrolling page. iOS doesn't provide any option for indicators besides the collection view's own scroll bar—arrows would likely look too similar to the table view's disclosure icon, which indicates tappability instead of scrollability in the parlance of iOS._ **Still think it's an issue even if Apple says it's fine. 09/09/2015**
- [ ] The little hand and people icons are hard to see because of their small size especially when there is more than one of them per country. _So, some context on this issue: our design requirements were to create a screen that allowed a broad overview of all countries, the goods in each, and the types of exploitation on each good. Fitting this much content on the screen to make the overview useful took many design discussions and iterations, and required shrinking the goods, and thus the exploitation symbols “on top” of them, to an appropriate size (12 pixels by 12 pixels) without losing their ability to be distinguished. To accomplish this, we carefully constructed the three exploitation symbols so that they would be as easy to see as possible at this scale. First, we made the two types of icons—the hand and the person—as differently shaped as possible, so that even if a user had poor eyesight, color-blindness, or a low resolution screen, the basic symbol still resolved to a distinct and recognizable shape: the hand becomes a circle, and the person becomes a diagonal slash. (We had to accomplish this with symbols that were also politically correct, which took a great deal of discussion and deliberation over many weeks.) We also color coded them so they are distinguishable by color—again, both for users with lower eyesight and for quick scanning—and tested the selected colors with colorblind users to make sure they could distinguish behind the two symbols. Though it was extremely difficult to satisfy all the requirements, we believe we accomplished it, and our testing confirmed this to our satisfaction._
- [ ] **RECOMMENDATION** - Should add the alphabetic list to the right as on other pages so people can skip down the list to the section they want. _This is a totally logical recommendation, and I mocked this up, but we decided that in practice, it doesn't really work for sections that are anything other than alphabetic. Instead of 26 one-letter sections that run neatly and narrowly down most of the side of the screen, you'd have only five or so very long section names.......iOS centers them vertically on the right side of the screen, pushing the rest of the content of the page left and creating massive whitespace along the entire right side of the screen above and below these 5 tiny options. We decided not to add a section index to any list that wasn't groups alphabetically._

Now as you complete correcting the issues listed above you can check them off, and we can test them again to see if that is correct. If something is out of scope you can add "**OUT OF SCOPE**" after the check box. It would be helpful to add the date the update was made to the beginning of the issue, so we know which ones were updated when.

**Comment by jfc3-dol** [comment]
Has any progress been made on this issue, since there are only a few more weeks before this project is to be completed.?

**Comment by skeeJay** [comment]
I've been out of the country and just returned to the office today. The plan is to resolve all open issues by early next week and get the app to you for final QA at that time, with an eye towards submission to the App Store 4–5 days later.

Today I will triage/answer the bullets noted in this ticket and either respond to each or open a new ticket to correct.

**Comment by jfc3-dol** [comment]
Thanks, for the update. If it makes sense to create other issues to keep track of thing that is fine. I will use our issue to keep tabs on what items have been completed.

No need to respond to my e-mail, which was only sent because I wasn't sure if you had seen this earlier or not.

**Comment by skeeJay** [comment]
Thanks, @jfc3-dol . I will create a separate issue for each bug we'll fix, yes. This will allow me to link each code commit to the bug it fixes in Github.

However, I'll note preliminarily that many of the issues here refer to standardized iOS controls. For example, elements of a table view such as disclosure indicators (">"), the section ("ABC…") index on the right, item separator lines, and section headings inside of a table view are provided for by the operating system—we haven't controlled the display, size, or color of any of these elements. Apple has designed these to be reusable and standardized across all iOS applications for recognizability and accessibility, and it's generally considered bad to customize or recreate standardized controls. I will note the items that this applies to in my triage.

**Comment by skeeJay** [comment]
@jfc3-dol , I’ve commented on all of these issues inline in the comment above. Many have been corrected in the new beta you’ll receive this week. Many others are comments on interface elements controlled by the operating system, which aren’t in our control. These are elements that Apple standardizes, including for accessibility purposes, so we can’t and shouldn’t mess with them. A couple referred to external websites which aren't part of this app, so I marked those "OUT OF SCOPE".

I’ve marked the other recommendations about design and will run them by the ILAB team at our weekly meeting this week. Anything we decided to update based on these recommendations are changes I can make with any other bugs the “official” testing uncovers later in the week.

The other accessibility issues I created tickets to correct:

* Make sector labels more visible on statistics screen https://github.com/USDepartmentofLabor/Child-Labor/issues/42
* Make green text more visible on indicator screens https://github.com/USDepartmentofLabor/Child-Labor/issues/41

Thanks very much, @jfc3-dol .

**Comment by jfc3-dol** [comment]
For now added a few comments about the PDF's being accessible might not be a project issue, but ILAB should fix those either way.

Once I get the newest application update from Joey I will go through these and check off the ones that have been corrected, along with adding a date when they were fixed of considered out of scope.

**Comment by jfc3-dol** [comment]
While looking at the latest version of the Child Labor iPhone application I found the following issues:

* [ ] The three new fact sheets in the "More Info" section seem to be PDF images, which VoiceOver will not be able to read. Plus, because they are in the application a person can not zoom in and out to enlarge the text so it is readable.
* [ ] 
* [ ] 
* [ ] 
* [ ] 
* [ ]

**Comment by jfc3-dol** [comment]
# New ILAB Child Labor iPhone Application Accessibility Issues (VoiceOver)
Here are the accessibility issues I have found so far with the new ILAB Child Labor iPhone application. I will add any other items I find in another comment later today.

## Home Screen
* [ ] The “**Countries**”, “**Goods**”, and “**Exploitation Types**” are not being read as buttons, so a user doesn’t know that they go to more data.

## More Info Screen
* [ ] The “**About this App**”, “**Methodology**”, “**Reports Fact Sheet**”, “***OCFT Fact Sheet**”, and “**Programs Fact Sheet**” text are not being read as buttons, because of this a user doesn’t know that they go to more data.

## Countries – A – Z - Screen
* [ ] When VoiceOver reads the list of countries, it does not tell a user that the country names are clickable so they can get more information. Guessing this should be set up as a button. 
* [ ] Is there a way to add what on a web page would be off-screen text to add to the individual lettered items. Doing this will allow VoiceOver to read **“B lettered country names**” where “**lettered country names**” is off-screen.
* [ ] There is no heading level one (h1) on this page, which should be “**Countries – A – Z**”.
* [ ] The individual letters should be a heading level two (h2) instead of only reading heading.

## Individual Country Screen
* [ ] The country name should be a heading level one (h1).
* [ ] How the country is doing (“**Moderate Advancement**”) should be a heading level two (h2).
* [ ] The next line with “**Number of Goods Produced with Exploitive Labor**” should be a heading level three (h3).
* [ ] When trying to read past the last item in the “**Goods**” sliding list I was not able to get to the next part of the text on the page. The problem could be a user error on my part and will have to look into how to get past it.
* [ ] Is there a way to make the “**Goods**” button be the image and text? Currently, it is reading the ALT text for the “**Goods**” such as broccoli twice. We need it there since the image is the button.
* [ ] Voiceover was reading the “**CL**” text, but would it be possible to add a legend on the page to tell what “**CL**” means or put off-screen text that says “**CL Child Labor**”?
* [ ] VoiceOver is not reading the little blue hand as anything. I was on the “**Guatemala**” individual country page.

**Comment by jfc3-dol** [comment]
# More Child Labor iPhone Application Accessibility Issues (VoiceOver)

## About this App Screen
* [ ] The text “About this App” should be a heading level one (h1).
* [ ] The text “What is this App?”, “How can I use this app?”, “What is the purpose of these reports?”, “Why isn’t Every Country in the world Represented here?”, and “Have ideas for ways to improve…” should all be a heading level two (h2).
* [ ] When reading this page, VoiceOver does not move to the next screen of text. It continues to read the whole page, but a person can’t following along if they are using VoiceOver to help them read because they have reading issues such as dyslexia, poor reading skills, etc.
* [ ] The e-mail address and website URL at the bottom of the page should have text in the paragraph instead of the e-mail address and URL, which is confusing to VoiceOver users. The text for the e-mail address should be “ Please **e-mail Global Kids** to with any idea for ways…”. The text for the URL should be “**please visit ILAB (Bureau of International Labor Affairs (ILAB))**”.

## Methodology Screen
* [ ] The text “Methodology” should be a heading level one (h1).
* [ ] When reading this page, VoiceOver does not move to the next screen of text. It continues to read the whole page, but a person can’t following along if they are using VoiceOver to help them read because they have reading issues such as dyslexia, poor reading skills, etc.

## Fact Sheet Screens (Reports, OFCT, and Programs)
* [ ] The URLs on these pages do not read as links and should use text and not the URL so people using VoiceOver can understand.
* [ ] There are no headings on these pages. The text under the first image should be a heading level one (h1) and the yellow, red, or green text for the different sections should be heading level two (h2).

## Countries – By Level - Screen
* [ ] When VoiceOver reads the list of countries, it does not tell a user that the country names are clickable so they can get more information. Guessing this should be set up as a button. 
* [ ] There is no heading level one (h1) on this page, which should be “Countries – By Level”.
* [ ] The text that looks like a heading ending in “Advancement” should be a heading level two (h2) instead of only being read as a heading.

## Countries – By Region - Screen
* [ ] When VoiceOver reads the list of countries, it does not tell a user that the country names are clickable so they can get more information. Guessing this should be set up as a button. 
* [ ] There is no heading level one (h1) on this page, which should be “Countries – By Region”.
* [ ] The different regions of the world should be a heading level two (h2) instead of only being read as a heading.

## Individual Countries Screen
* [ ] Any reason China, Burma, Malaysia, and possibly more countries do not have any text under the list of “Goods Produced with Exploitive Labor” slider. Also missing is the “Suggested Actions”, Statistics”, “International Conventions”, and “Legal Standards” buttons. On other pages, these are not being announced as buttons on other pages.
* [ ]  VoiceOver is reading the text that is hidden visually off-screen before the “More” button.
* [ ] Had trouble with other countries when trying to read past the last item in the “Goods” sliding list. I was not able to get to the next part of the text on the page. Maybe this could be a user error on my part. After looking into it further it seems to work for most of the lists if goods under the "Exploitation Types", so most likely an issue on the country screens.
* [ ] The “Suggested Actions”, “Statistics”, “International Conventions”, and “Legal Standards” text are not being read as buttons, so the user knows that there is more information on another screen.

## Individual Countries - Suggested Actions - Screen
* [ ] There are no properly defined headings. The text “Suggested Actions” should be a heading level one (h1) and the other headings below should be heading level two (h2).
* [ ] Why is the text “No Actions” under the heading level two (h2) background color the same color as the headings?

## Individual Countries – Statistics - Screen
* [ ] There are no properly defined headings. The text “Statistics” should be a heading level one (h1) and the other headings below should be heading level two (h2).

## Individual Countries – International Conventions - Screen
* [ ] There are no properly defined headings. The text “International Conventions” should be a heading level one (h1) and the other headings below should be heading level two (h2).

## Individual Countries – Legal Standards - Screen
* [ ] There are no properly defined headings. The text “Legal Standards” should be a heading level one (h1).

## Goods – A – Z - Screen
* [ ] All the “Goods” text are not being read as buttons or whatever they need to be so user knows that they have more information on another screen.
* [ ] Need a heading level one (h1) which should be “Goods A – Z” and then the other headings below should be a heading level two (h2).

## Goods – By Sector – Screen
* [ ] All the “Goods” text are not being read as buttons or whatever they need to be so user knows that they have more information on another screen.
* [ ] Need a heading level one (h1) which should be “Goods By Sector” and then the other headings like “Agriculture/forestry/Fishing” or “Manufacturing”, etc. below should be a heading level two (h2).

## Individual Goods Screen
* [ ] The “Goods” name should be a heading level one (h1).
* [ ] What sector (“Manufacturing”) should be a heading level two (h2).
* [ ] The text “Produced with…” should be a heading level three (h3).
* [ ] When VoiceOver reads the country name, it is not notified that there is more information about that country on another screen.
* [ ] While looking at “Bidis (hand rolled Cigarettes)” and I chose “Forced Child Labor” the heading below said there were zero items, and then it read the country names. When I tried tapping on this nothing happened and the color contrast got even lighter that it was already.
* [ ] Why when VoiceOver reads the individual lines with the country name, and the types of forced labor does VoiceOver read the type of forced labor first and then the country name? Everywhere else in this application we read things from left to right, and now it's right to left.

## Exploitation Type Screen
* [ ] VoiceOver will not read past “Angola diamond button” or “Honduras” when I’m trying to read the entire screen.
* [ ] When a country has more than seven or eight goods/exploitation types it skips over the rest of the list for some reason. At least the first few countries had this happen such as Brazil, Burma, Cambodia, etc. 
* [ ] Question: Why are the exploited goods not in alphabetic order?

**Comment by spoonstein** [comment]
@jfc3-dol you can't setup iOS screens like you can HTML. So there isn't a h1, h2, h3 kind of layering. Do you have something that reflects accessibility for iOS? This is what I see: https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/iPhoneAccessibility/Accessibility_on_iPhone/Accessibility_on_iPhone.html#//apple_ref/doc/uid/TP40008785-CH100-SW1

**Comment by jfc3-dol** [comment]
There are screens in this application that have different heading levels such as the "About this App" under the more information section.

**Comment by skeeJay** [comment]
A small number of text-only screens in the app, "About this App" and "Methodology," are web views, meaning they contain HTML content and do indeed have HTML elements. But all other screens use native iOS controls such as label views and table views (defined at https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/MobileHIG/Bars.html#//apple_ref/doc/uid/TP40006556-CH12-SW1), and HTML elements aren't applicable in them.

**Comment by jfc3-dol** [comment]
Okay. 

Have never read about or built an iOS application so was trying to use terminology I understand from the website side of things.

In a few cases/screens, I think we still need to add a heading here and there to make it easier to navigate the application.

**Comment by jfc3-dol** [comment]
@skeeJay are you making any progress with all the open accessibility, usability, and VoiceOver issues? If not are they going to be worked on while Apple is reviewing the application since most if not all of them need to be fixed before the release or very soon after? If DOL is doing to put the application out with these outstanding issues, we (OPA DEC) will need someone in ILAB taking responsibility for any open issues.

**Comment by jfc3-dol** [comment]
Closing this issue because a lot has changed in the last two years and some might not apply and other could be fixed.

Closing issue.
